src/common/contexts/provider.tsx

```diff
@@ -83,105 +81,78 @@ export const ProviderContextProvider: FunctionComponent<ProviderContextProviderP
+  );
 
-  const updateProviderOrSigner = async (newProviderOrSigner: typeof providerOrSigner) => {
-    try {
-      if (!Signer.isSigner(newProviderOrSigner)) {
-        setIsConnected(false);
-      } else {
-        await (newProviderOrSigner as Signer).getAddress();
-        setIsConnected(true);
-      }
-    } catch (e) {
-      setIsConnected(false);
-    }
-    setProviderOrSigner(newProviderOrSigner);
-  };
+  const [providerType, setProviderType] = useState<SIGNER_TYPE>(SIGNER_TYPE.IDENTITY);
+  const [currentChainId, setCurrentChainId] = useState<ChainId | undefined>(
+    isSupportedNetwork(defaultChainId) ? defaultChainId : undefined
+  );
+  const [account, setAccount] = useState<string | undefined>();
+  const [providerOrSigner, setProviderOrSigner] = useState<providers.Provider | ethers.Signer | undefined>(
+    defaultProvider.current
+  );
+  const [provider, setProvider] = useState<providers.Provider | undefined>(defaultProvider.current);
 
   const changeNetwork = async (chainId: ChainId) => {
-    if (!isSupportedNetwork(chainId)) throw new UnsupportedNetworkError(chainId);
-
-    const chainInfo = getChainInfo(chainId);
-
-    try {
-      const web3provider = getWeb3Provider();
-      await requestSwitchChain(chainId);
-      await updateProviderOrSigner(web3provider.getSigner());
-    } catch (e: unknown) {
-      if (e instanceof NoMetaMaskError) {
-        console.warn(e.message);
-        await updateProviderOrSigner(createProvider(chainInfo.chainId));
-      } else {
-        throw e;
-      }
-    }
+    await walletSwitchChain(chainId);
+    setCurrentChainId(chainId);
   };
 
-  const getProvider = useCallback(() => {
-    if (providers.Provider.isProvider(providerOrSigner)) return providerOrSigner;
-    if (Signer.isSigner(providerOrSigner)) return providerOrSigner.provider;
-    return undefined;
-  }, [providerOrSigner]);
-
-  const getSigner = useCallback(
-    () => (isConnected ? (providerOrSigner as Signer) : undefined),
-    [isConnected, providerOrSigner]
-  );
-
-  const getTransactor = useCallback(() => getSigner() ?? getProvider(), [getProvider, getSigner]);
-
-  const getWeb3Provider = () => {
+  const updateProvider = useCallback(async () => {
     const { ethereum, web3 } = window;
     const metamaskExtensionNotFound = typeof ethereum === "undefined" || typeof web3 === "undefined";
-    if (metamaskExtensionNotFound || !ethereum.request) throw new NoMetaMaskError();
-
-    const injectedWeb3 = ethereum || web3.currentProvider;
-    if (!injectedWeb3) throw new Error("No injected web3 provider found");
-    return new ethers.providers.Web3Provider(injectedWeb3, "any");
-  };
+    if (metamaskExtensionNotFound || !ethereum.request) {
+      setProvider(createProvider(currentChainId || defaultChainId));
+      setAccount(undefined);
+    } else {
+      const injectedWeb3 = ethereum || web3.currentProvider;
+      if (!injectedWeb3) throw new Error("No injected web3 provider found");
```

Comment:
should the throw error be move to above? above condition has a check for `web3`

Reply:
Ahh! Good point! I have added a check to make sure `web3` is available.

Reply:
oh i meant the `metamaskExtensionNotFound` part already check for undefined. i thought the `injectedWeb3` check wasn't necessary

Reply:
Ohh ya!! You are right! It's not needed. Removed it in https://github.com/TradeTrust/tradetrust-website/pull/578/commits/ad131d54b93c0905f9db5ec80bbf6c3e247b480d.